.github/workflows/e2e-tests-reusable-workflow.yaml

```diff
@@ -32,6 +32,13 @@ jobs:
         - name: Setup and start KinD cluster
           uses: ./.github/workflows/actions/kind
 
+        - name: Install Ray
+          if: inputs.plugin-test
+          run: |
+            pip install -U "ray[default]"
```

Comment:
I think we should pin the version of Ray to a specific version in CI.

Reply:
Is there a specific version that we should be using? 

Also just to clarify since I'm not too familiar with .github workflows. By pinning a specific Ray version do you just mean something like this: `pip install -U "ray[default]==2.10.0"?

Reply:
Yes. I think we can use the latest version `2.40.0`. It's better to pin dependency versions in CI.